### PR TITLE
resource/aws_rds_cluster_instance: Support Performance Insights

### PR DESCRIPTION
**Pull request by kwent**
## Support Performance Insights for rds_cluster_instance

Fixes: https://github.com/terraform-providers/terraform-provider-aws/issues/2198

```
$ go test -run=TestAccAWSRDSClusterInstance_withInstancePerformanceInsights -timeout=30s -parallel=4 ./aws
ok  	github.com/terraform-providers/terraform-provider-aws/aws	0.056s
```

### CODE REVIEW

**Comment by kwent** [comment]
No idea why `EnablePerformanceInsights` is not available ? Is the aws-sdk-go up to date ?

```
aws/resource_aws_rds_cluster_instance.go:359:7: db.EnablePerformanceInsights undefined (type *rds.DBInstance has no field or method EnablePerformanceInsights)
aws/resource_aws_rds_cluster_instance.go:360:43: db.EnablePerformanceInsights undefined (type *rds.DBInstance has no field or method EnablePerformanceInsights)
```

**Comment by kwent** [comment]
Any reviewers for this ?

**Comment by kwent** [comment]
@Ninir Thanks. Yes probably ! I will fix the test.

Should i also check against the engine and notify the user that Performance Insights is only available for `engine` = `aurora-postgresql` or just let the API threw this error ?

**Comment by kwent** [comment]
Hi @Ninir,

The PR should be in a much better place now. Let me know !

**Comment by kwent** [comment]
Any follow up on this ? cc @Ninir 

How can i help ?

Regards

**Comment by Ninir** [comment]
Hi @kwent 

Just ran your tests, and got:

```
--- FAIL: TestAccAWSRDSClusterInstance_withInstancePerformanceInsights (117.00s)
	testing.go:503: Step 0 error: Error applying: 1 error(s) occurred:
		
		* aws_rds_cluster_instance.cluster_instances: 1 error(s) occurred:
		
		* aws_rds_cluster_instance.cluster_instances: InvalidParameterCombination: Cannot find version 9.6.3 for aurora
```

Can you investigate please? thanks!

**Comment by kwent** [comment]
I believe i now fixed it. I followed https://github.com/terraform-providers/terraform-provider-aws/issues/2419 recommendation.

**Comment by Ninir** [comment]
Hi @kwent !

Sure, here is the result:

```
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSRDSClusterInstance_withInstancePerformanceInsights'

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccAWSRDSClusterInstance_withInstancePerformanceInsights -timeout 120m
=== RUN   TestAccAWSRDSClusterInstance_withInstancePerformanceInsights
--- FAIL: TestAccAWSRDSClusterInstance_withInstancePerformanceInsights (116.18s)
	testing.go:503: Step 0 error: Error applying: 1 error(s) occurred:
		
		* aws_rds_cluster_instance.cluster_instances: 1 error(s) occurred:
		
		* aws_rds_cluster_instance.cluster_instances: InvalidParameterCombination: RDS does not support creating a DB instance with the following combination: DBInstanceClass=db.r3.large, Engine=aurora-postgresql, EngineVersion=9.6.3, LicenseModel=postgresql-license. For supported combinations of instance class and database engine version, see the documentation.
			status code: 400, request id: c6cfaa29-b903-4322-8cae-bb12b56a233f
```

Are you able to run those tests on your own? Tell me if you need any help on that :)
It's getting ready to cross to the finish line! 🚀

File: aws/resource_aws_rds_cluster_instance_test.go [hunk 1]
```diff
@@ -505,3 +529,65 @@ resource "aws_db_parameter_group" "bar" {
...
+  master_password    = "mustbeeightcharaters"
+  storage_encrypted = true
+  skip_final_snapshot = true
+}
+
+resource "aws_rds_cluster_instance" "cluster_instances" {
+  engine                  = "aurora-postgresql"
+  identifier              = "tf-cluster-instance-%d"
+  cluster_identifier      = "${aws_rds_cluster.default.id}"
+  instance_class          = "db.r3.large"
```
Comment (ColinHebert) [inline aws/resource_aws_rds_cluster_instance_test.go]: Postgresql only works with r4 instances: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.DBInstanceClass.html

Comment (kwent) [reply aws/resource_aws_rds_cluster_instance_test.go]: @ColinHebert Thanks ! Fixed.

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.7.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#170-january-12-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**